### ISSUE TITLE
General feedback on kestrel connection metrics

### ISSUE THREAD

**Issue by lmolkova** [dotnet/aspnetcore#52437]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

Not related to any functional problem.

### Describe the solution you'd like

Several [kestrel connection metrics](https://github.com/open-telemetry/semantic-conventions/blob/e95fde026557e42fa832887b416b772101921986/docs/dotnet/dotnet-kestrel-metrics.md) could be grouped together to better align with otel semantics.

See original discussion in https://github.com/open-telemetry/semantic-conventions/pull/283/files/e95fde026557e42fa832887b416b772101921986#r1400552625

TL;DR:

`kestrel.active_connections`, `kestrel.queued_connections` could be measured with something like `kestrel.current.connections` or `kestrel.connections` and additional attribute like `kestrel.connection.state` = `active` | `queued`.

`kestrel.upgraded_connections` might also be measured with the same counter using an attribute that captures original protocol version (e.g. `http.protocol.initial_version`).

**Comment by JamesNK** [comment]
> `kestrel.active_connections`, `kestrel.queued_connections` could be measured with something like `kestrel.current.connections` or `kestrel.connections` and additional attribute like `kestrel.connection.state` = `active` | `queued`.

How would a connection go from queued to active? Would you need to subtract it and then add it again?

e.g.
Connection is queued: `kestrel.connections` `state=queued` + 1
Connection is ready: `kestrel.connections` `state=queued` - 1 AND `kestrel.connections` `state=active` + 1
Connection is finished: `kestrel.connections` `state=active` - 1
